src/Integration/Connection/ConnectionController.cs

```diff
@@ -183,7 +183,15 @@ private bool OnDontWarnAgainStatus()
         ConnectionInformation IConnectionInformationProvider.GetConnectionInformation(ConnectionInformation currentConnection)
         {
             var dialog = new ConnectionInformationDialog();
-            return dialog.ShowDialog(currentConnection);
+            var connectionInfo = dialog.ShowDialog(currentConnection);
+
+            if (connectionInfo != null)
+            {
+                // User has not specified the organization yet. Read it from existing connection, not
+                // to prompt again (if the organization is set already, it means the binding exists).
+                connectionInfo.Organization = currentConnection?.Organization;
```

Comment:
Are there any circumstances in which the user would want to change the organisation, and would this change prevent them from being able to do it?
It does seem weird that we are prompting the user for only some of the connection information. If we need to prompt, why aren't we prompting for everything?

Reply:
The questions do make sense but looking at the history we have never supported the organization change so I wouldn't make any fix on that yet (definitely worth creating the ticket though).

Reply:
I've done some testing and this part of the fix doesn't seem to be required - the change in _BoundSonarQubeProjectExtensions.cs_ is enough to re-connect without prompting when the solution is re-opened. I didn't actually hit this code path when re-opening or updating a bound solution.
If the change isn't needed to fix the bug then it would be safer to remove it in case it has an impact in other scenarios.